server/server.go

```diff
@@ -159,7 +159,8 @@ func (server *Server) getDatabaseConfig() (string, string, bool, bool) {
 	}
 	databaseDropOnCreate := config.GetBool("database/drop_on_create", false)
 	databaseCascade := config.GetBool("database/cascade_delete", false)
-	return databaseType, databaseConnection, databaseDropOnCreate, databaseCascade
+	databaseAutoMigrate := config.GetBool("database/auto_migrate", true)
```

Comment:
There is a WIP including the mentioned sanity checks:
https://github.com/przemyslaw-dobrowolski-cl/gohan/tree/migration
It contains a part of migration code that checks the DB during startup and if there are any changes in DB, the startup is aborted. Maybe we can wait for it to be merged first and then we can continue with the change of the default value of no_init flag?